lib/spack/spack/directives.py

```diff
@@ -180,24 +180,43 @@ def _depends_on(pkg, spec, when=None):
         when = pkg.name
     when_spec = parse_anonymous_spec(when, pkg.name)
 
+    if deptypes is None:
+        # The default deptype is build and link because the common case is to
+        # build against a library which then turns into a runtime dependency
+        # due to the linker.
```

Comment:
An easier way to think about this is whether, when `A` -> `B`, `B` needs to be in the `PATH` when you `module load A` in order for `A` to work.